types/lib/nodecg-instance.d.ts

```diff
@@ -0,0 +1,106 @@
+import {IRouter, RequestHandler} from 'express-serve-static-core';
+
+import {Logger} from './logger';
+import {ReplicantOptions, Replicant} from './replicant';
+import {NodeCGConfig} from './config';
+import {SendMessageReturnType} from './nodecg-static';
+import {Platform} from './platform';
+
+interface NodeCGCommon<P extends Platform, M = SendMessageReturnType<P>> {
+	bundleName: string;
+	bundleConfig: any;
+	bundleVersion: string;
+	readonly bundleGit: {
+		branch: string;
+		hash: string;
+		shortHash: string;
+		date?: Date;
+		message?: string;
+	};
+	Logger: typeof Logger;
+	log: Logger;
+	readonly config: NodeCGConfig;
+	sendMessage(
+		messageName: string,
+		cb?: (error: any, ...args: any[]) => void
+	): M;
+	sendMessage(
+		messageName: string,
+		data: any,
+		cb?: (error: any, ...args: any[]) => void
+	): M;
+	listenFor(messageName: string, handlerFunc: (message: any) => void): void;
+	listenFor(
+		messageName: string,
+		bundleName: string,
+		handlerFunc: (message: any) => void
+	): void;
+	unlisten(messageName: string, handlerFunc: (message: any) => void): void;
+	unlisten(
+		messageName: string,
+		bundleName: string,
+		handlerFunc: (message: any) => void
+	): void;
+	Replicant<V>(name: string, opts?: ReplicantOptions<V>): Replicant<V, P>;
+	Replicant<V>(
+		name: string,
+		namespace: string,
+		opts?: ReplicantOptions<V>
+	): Replicant<V, P>;
+}
+
+/**
+ * NodeCG instance in extensions
+ */
+export interface NodeCGServer extends NodeCGCommon<'server'> {
+	getSocketIOServer(): SocketIO.Server;
+	mount: IRouter['use'];
+	util: {
+		authCheck: RequestHandler;
+	};
+	extensions: {
+		[bundleName: string]: (nodecg: NodeCGServer) => void;
+	};
+}
+
+/**
+ * NodeCG instance in browser
+ */
+export interface NodeCGBrowser extends NodeCGCommon<'browser'> {
+	socket: SocketIOClient.Socket;
+	soundReady?: boolean;
+	getDialog(
+		name: string,
+		bundle?: string
+	): ReturnType<ParentNode['querySelector']>;
+	getDialogDocument(name: string, bundle?: string): Document;
+	findCue(cueName: string): Cue | undefined;
+	playSound(
+		cueName: string,
+		opts?: {updateVolume?: boolean}
+	): createjs.AbstractSoundInstance;
```

Comment:
This line also causes an error for me duration compilation:
```
../../types/lib/nodecg-instance.d.ts:81:5 - error TS2503: Cannot find namespace 'createjs'.

81  ): createjs.AbstractSoundInstance;
       ~~~~~~~~
```

Reply:
Installing `@types/soundjs` from npm should fix it. They are included in package.json of this PR.

Reply:
It is declared as global namespace once the @type package is installed. But as typedef library it had to use triple slash directives (`/// <reference/>`) to work.

Does it work without errors now?

Reply:
The SocketIO one is fixed, and the createjs one now produces a different error:
```
../../types/lib/nodecg-instance.d.ts:95:14 - error TS2694: Namespace 'createjs' has no exported member 'AbstractSoundInstance'.

95  ): createjs.AbstractSoundInstance;
                ~~~~~~~~~~~~~~~~~~~~~
```

Reply:
I referenced wrong module with same namespace name. This is why global namespace is a bad idea and being replaced with the ES6 module way. 🤕 

I'm looking into why the test is fine but it doesn't work for bundles. I'll try making npm project in test directory so it simulates bundle's directory. If you know better ways, let me know.

Reply:
All good now after that latest commit 👍

Reply:
Glad to hear :)